@AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fcr/fn_fcrLongbow.sqf

```diff
@@ -10,42 +10,65 @@ Returns:
 Examples:
     [_heli] spawn fza_fnc_fcrLongbow;
 Author:
-    Rosd6(Dryden), BradMick
+    Rosd6(Dryden)
 ---------------------------------------------------------------------------- */
 if (!(isNil "fza_ah64_nofcr")) exitwith {};
 _heli = vehicle player;
+if !((vehicle player) animationphase "plt_apu" > 0.5 || (isEngineOn _heli)) exitwith {};
 
 #define AGMODE_GND 0
 #define AGMODE_AIR 1
-_dataLinkArray = [];
+_datalinkarray = [];
 
-if ((vehicle player) animationphase "plt_apu" > 0.5 || (isEngineOn _heli)) then {
-	if (isVehicleRadarOn _heli && (_heli animationPhase "fcr_enable" == 1) && _heli getHit "radar" < 0.8) then {
-        _dataLinkArray = (listRemoteTargets playerSide select {_x # 1 > 6}) apply {_x # 0};
-		{
-			if (alive _x) then {
-				_distOffAxis = abs ([[_heli, (getposatl _heli select 0), (getposatl _heli select 1), (getposatl _x select 0), (getposatl _x select 1)] call fza_fnc_relativeDirection] call CBA_fnc_simplifyAngle180);
+if (isVehicleRadarOn _heli && (_heli animationPhase "fcr_enable" == 1) && _heli getHit "radar" < 0.8) then {
+	_datalinkarray = (listRemoteTargets playerSide) apply {_x # 0};
+	{
+		_distOffAxis = abs ([[_heli, (getposatl _heli select 0), (getposatl _heli select 1), (getposatl _x select 0), (getposatl _x select 1)] call fza_fnc_relativeDirection] call CBA_fnc_simplifyAngle180);
+		if (_x == _heli || _x isKindOf "man" || _x isKindOf "StaticCannon" || laserTarget _heli == _x) then {
+			_dataLinkArray = _dataLinkArray - [_x];
+		};
+		if !(alive _x) then {
+			_dataLinkArray = _dataLinkArray - [_x];
+		};
+		if ((_heli getVariable "fza_ah64_agmode" == AGMODE_GND) && (_distOffAxis > 45)) then {
+			_dataLinkArray = _dataLinkArray - [_x];
+		};				
+		if (_heli getVariable "fza_ah64_agmode" == AGMODE_AIR && !(_x isKindOf "air")) then {
+			_dataLinkArray = _dataLinkArray - [_x];
+		};
 
-				if (_x == _heli || _x isKindOf "man" || _x isKindOf "StaticCannon" || laserTarget _heli == _x) then {
-					_dataLinkArray = _dataLinkArray - [_x];
-				};
-				if ((_heli getVariable "fza_ah64_agmode" == AGMODE_GND) && (_distOffAxis > 45)) then {
-					_dataLinkArray = _dataLinkArray - [_x];
+		//LOS check Beg
+		if (fza_ah64_ExperimentalFCR) then {
+			_lastPos = _heli modelToWorldWorld [0,2.1,2.2];
```

Comment:
Rather than using a hard-coded point in the code, could you use a memory point? I know currently we have a freeze on model changes, but for now you could use `vehicle player selectionPosition "velka osa"`, which gives a close enough value at `[0.000248395,2.06466,0.962408]`. Then, in the future when we can add a memory point, we make a `"radar"` one or something like that.

Reply:
is there a performance difference between these to methods?

Reply:
For the sake of having a correct fcr simulation over readability until the radar mempoint is added